[Lens] Remove all soClient logic and type references

nickofthyme opened the pull request:
## Summary

Remove all references to `soClient`. Most were lingering types but a few need core refactoring.

### Files to fix

- [x] `x-pack/platform/plugins/shared/lens/public/persistence/saved_objects_utils/find_object_by_title.test.ts`
- [x] `x-pack/platform/plugins/private/graph/public/helpers/saved_objects_utils/find_object_by_title.test.ts`
- [x] `src/platform/plugins/shared/visualizations/public/utils/saved_objects_utils/find_object_by_title.test.ts`
- [x] `src/platform/plugins/private/event_annotation/public/event_annotation_service/service.test.ts`
- [x] `src/platform/plugins/shared/vis_types/timeseries/public/application/contexts/query_input_bar_context.ts`
- [x] `src/platform/plugins/shared/visualizations/public/mocks.ts`
- [x] `src/platform/plugins/shared/visualizations/public/plugin.ts`
- [x] `src/platform/plugins/shared/visualizations/public/services.ts`
- [x] `src/platform/plugins/shared/visualizations/public/types.ts`
- [x] `src/platform/plugins/shared/visualizations/public/utils/saved_objects_utils/save_with_confirmation.test.ts`
- [x] `src/platform/plugins/shared/visualizations/public/utils/saved_objects_utils/save_with_confirmation.ts`
- [x] `src/platform/plugins/shared/visualizations/public/vis_types/vis_type_alias_registry.ts`
- [x] `src/platform/plugins/shared/visualizations/public/visualize_app/components/visualize_listing.tsx`
- [x] `src/platform/plugins/shared/visualizations/public/wizard/search_selection/show_saved_object.test.ts`
- [x] `x-pack/platform/plugins/private/graph/public/components/search_bar.test.tsx`
- [x] `x-pack/platform/plugins/private/graph/public/helpers/saved_objects_utils/save_with_confirmation.ts`
- [x] `x-pack/platform/plugins/private/graph/public/helpers/use_workspace_loader.ts`
- [x] `x-pack/platform/plugins/private/graph/public/types/app_state.ts`
- [x] `x-pack/platform/plugins/shared/lens/public/types.ts`

closes #224307

### Checklist

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [x] This was checked for breaking HTTP API changes, and any breaking changes have been approved by the breaking-change committee. The `release_note:breaking` label should be applied in these situations.

nickofthyme committed:
chore: remove all soClient type references

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-renovate-helper --> Click to trigger **kibana-renovate-helper** for this PR!

nickofthyme committed:
chore: fix event anno service

nickofthyme committed:
chore: fix missing vis SO types

nickofthyme committed:
Merge branch 'main' into rm-so-client-lens

nickofthyme committed:
chore: cleanup visualizations plugin types and code

- fix missed type changes
- fix title dup check logic
- add title dup check to embeddable
- make title dup check optional on save
- remove fallback title dup confirm modal

nickofthyme committed:
chore: cleanup lens plugin types to remove SO types

nickofthyme committed:
chore: cleanup visualizations plugin types and code

- fix missed type changes
- rename savedObjectStore to lensDocumentService
- move checkForDuplicateTitle under lensDocumentService
- remove need for services in checkForDuplicateTitle
- remove unused savedObjectStore from Lens app props
- fix title dup check logic
- make onTitleDuplicate required
- remove fallback title dup confirm modal

nickofthyme commented on x-pack/platform/plugins/shared/lens/public/persistence/lens_document_service.ts:
```diff
@@ -0,0 +1,128 @@
+   * returns Promise<true> when it's no duplicate, or the modal displaying the warning
+   * that's there's a duplicate is confirmed, else it returns a rejected Promise<ErrorMsg>
+   */
+  async checkForDuplicateTitle(
+    {
+      id,
+      title,
+      isTitleDuplicateConfirmed,
+      lastSavedTitle,
+      copyOnSave,
+    }: CheckDuplicateTitleOptions,
+    onTitleDuplicate: () => void
+  ): Promise<boolean> {
```
Moved this from independent function in `check_for_duplicate_title.ts`

nickofthyme commented on x-pack/platform/plugins/shared/lens/public/persistence/saved_objects_utils/display_duplicate_title_confirm_modal.ts:
```diff
@@ -1,35 +0,0 @@
-import { SAVE_DUPLICATE_REJECTED } from './constants';
-import { confirmModalPromise } from './confirm_modal_promise';
-
-export function displayDuplicateTitleConfirmModal(
```
No longer needed, handled by `onTitleDuplicate`

nickofthyme reviewed (commented):
Lens changes to title duplication check

nickofthyme commented on src/platform/plugins/shared/visualizations/public/embeddable/visualize_embeddable.tsx:
```diff
@@ -306,7 +307,18 @@ export const getVisualizeEmbeddableFactory: (deps: {
       },
       canLinkToLibrary: () => Promise.resolve(!linkedToLibrary),
       canUnlinkFromLibrary: () => Promise.resolve(linkedToLibrary),
-      checkForDuplicateTitle: () => Promise.resolve(), // Handled by saveToLibrary action
+      checkForDuplicateTitle: async (
+        newTitle: string,
+        isTitleDuplicateConfirmed: boolean,
+        onTitleDuplicate: () => void
+      ) => {
+        await checkForDuplicateTitle(
+          { title: newTitle, lastSavedTitle: '' },
+          false,
+          isTitleDuplicateConfirmed,
+          onTitleDuplicate
+        );
+      },
```
Yes this **was** being handled in the `saveToLibrary` method but it was problematic.

The `dashboard` plugin only passed the `onTitleDuplicate` callback to the `embeddable.checkForDuplicateTitle`.

https://github.com/elastic/kibana/blob/94c0008ea6f29c60fee0013447c87d48244ee9b7/src/platform/plugins/shared/dashboard/public/dashboard_actions/library_add_action.tsx#L94-L100

Because of this the vis plugin would have to use the fallback confirm modal when a duplicate title is found. Even then this would break and just be left in an infinite loading loop.

But this duplicate check should be called when saving from the vis app itself. This now handled the dup check at the dashboard level and only check the duplicate title on save when `onTitleDuplicate` is defined.

nickofthyme commented on src/platform/plugins/shared/visualizations/public/utils/saved_objects_utils/check_for_duplicate_title.ts:
```diff
@@ -39,18 +33,13 @@ export async function checkForDuplicateTitle(
   }
 
-  if (onTitleDuplicate) {
-    onTitleDuplicate();
-    return Promise.reject(new Error(SAVE_DUPLICATE_REJECTED));
-  }
+  onTitleDuplicate();
 
-  // TODO: make onTitleDuplicate a required prop and remove UI components from this class
-  // Need to leave here until all users pass onTitleDuplicate.
-  return displayDuplicateTitleConfirmModal(savedObject, services);
```
`onTitleDuplicate` is now always defined.

nickofthyme commented on src/platform/plugins/shared/visualizations/public/utils/saved_objects_utils/display_duplicate_title_confirm_modal.ts:
```diff
@@ -1,43 +0,0 @@
-import { SAVE_DUPLICATE_REJECTED } from './constants';
-import { confirmModalPromise } from './confirm_modal_promise';
-
-export function displayDuplicateTitleConfirmModal(
```
No more need for this fallback UI.

nickofthyme commented on src/platform/plugins/shared/visualizations/public/utils/saved_visualize_utils.ts:
```diff
@@ -359,20 +359,27 @@ export async function saveVisualization(
-      migrationVersion: savedObject.migrationVersion,
-      references: extractedRefs.references,
-    };
+    if (onTitleDuplicate) {
+      // Only checks dups if onTitleDuplicate is passed.
+      // The save action is done from multiple places. In some cases like the embeddable,
+      // the title dup check is done before save is called and in other places we need to check.
+      await checkForDuplicateTitle(
+        savedObject,
+        copyOnSave,
+        isTitleDuplicateConfirmed,
+        onTitleDuplicate
+      );
+    }
```
Only check the duplicate title on save when `onTitleDuplicate` is defined. Avoids calling when saving from the dashboard.

nickofthyme commented on src/platform/plugins/shared/visualizations/public/utils/saved_visualize_utils.ts:
```diff
@@ -359,20 +359,27 @@ export async function saveVisualization(
-      onTitleDuplicate,
-      services
-    );
-    const createOpt = {
-      migrationVersion: savedObject.migrationVersion,
-      references: extractedRefs.references,
-    };
```
These options are never typed so old props are never culled. `migrationVersion` is no longer supported.

nickofthyme commented on src/platform/plugins/shared/visualizations/public/visualize_app/components/visualize_listing.tsx:
```diff
@@ -209,12 +209,10 @@ const useTableListViewProps = (
                       id,
                       title: value,
                       lastSavedTitle: content.title,
-                      getEsType: () => content.type,
```
Never used.

nickofthyme commented on src/platform/plugins/shared/visualizations/public/visualize_app/components/visualize_listing.tsx:
```diff
@@ -209,12 +209,10 @@ const useTableListViewProps = (
                     false,
                     false,
-                    () => {},
-                    startServices
```
No longer need `startServices`

nickofthyme reviewed (commented):
Vis plugin related changes to title duplication logic

nickofthyme committed:
chore: remove console log

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/yarn_deduplicate'

nickofthyme committed:
Merge branch 'main' into rm-so-client-lens

nickofthyme committed:
chore: prune unused i18n strings

nickofthyme committed:
chore: fix more tests and type errors

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/317479)
* Commit: 04bd9d1e71270ea5027a28cc92e14f017c2f7b0a

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/317479#0197f0d1-667d-45ae-92c7-2a009543fe7d)

### History
* :broken_heart: [Build #315611](https://buildkite.com/elastic/kibana-pull-request/builds/315611) failed d336259097296a69bb0e639850bd0c1cb7b80105

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/317479","number":317479,"commit":"04bd9d1e71270ea5027a28cc92e14f017c2f7b0a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/315611","number":315611,"commit":"d336259097296a69bb0e639850bd0c1cb7b80105"}],"number":317479}
buildkite-pr-comment-->

nickofthyme committed:
chore: another type error

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/317480)
* Commit: 9a7959558e6deda20f8e917bf41ce30bcc584b64

### Failed CI Steps
* [Jest Tests #2](https://buildkite.com/elastic/kibana-pull-request/builds/317480#0197f101-d0e9-411c-a5a8-d3ed9be44a69)
* [Jest Tests #2](https://buildkite.com/elastic/kibana-pull-request/builds/317480#0197f125-2a77-4ec8-95ad-fc9c0b101a06)
* [FTR Configs #87](https://buildkite.com/elastic/kibana-pull-request/builds/317480#0197f101-d13b-4e96-9e07-332d6ef64302)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/317480#0197f101-d0e9-411c-a5a8-d3ed9be44a69) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/317480/jobs/0197f101-d0e9-411c-a5a8-d3ed9be44a69/artifacts/0197f125-18e3-4c12-8502-095027c47a76) Jest Tests #<span></span>2 / saved_visualize_utils saveVisualization isTitleDuplicateConfirmed as false we should not save vis with duplicated title
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/317480#0197f125-2a77-4ec8-95ad-fc9c0b101a06) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/317480/jobs/0197f125-2a77-4ec8-95ad-fc9c0b101a06/artifacts/0197f12a-4974-4c74-adda-cfafdd1a6d50) Jest Tests #<span></span>2 / saved_visualize_utils saveVisualization isTitleDuplicateConfirmed as false we should not save vis with duplicated title
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/317480#0197f101-d0e9-411c-a5a8-d3ed9be44a69) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/317480/jobs/0197f101-d0e9-411c-a5a8-d3ed9be44a69/artifacts/0197f125-18e3-4f6a-be4f-2bd01f2d9f50) Jest Tests #<span></span>2 / saved_visualize_utils saveVisualization isTitleDuplicateConfirmed as true we should save vis with duplicated title
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/317480#0197f125-2a77-4ec8-95ad-fc9c0b101a06) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/317480/jobs/0197f125-2a77-4ec8-95ad-fc9c0b101a06/artifacts/0197f12a-4975-4f1e-9852-b1917706bc29) Jest Tests #<span></span>2 / saved_visualize_utils saveVisualization isTitleDuplicateConfirmed as true we should save vis with duplicated title

### History
* :broken_heart: [Build #317479](https://buildkite.com/elastic/kibana-pull-request/builds/317479) failed 04bd9d1e71270ea5027a28cc92e14f017c2f7b0a
* :broken_heart: [Build #315611](https://buildkite.com/elastic/kibana-pull-request/builds/315611) failed d336259097296a69bb0e639850bd0c1cb7b80105

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/317480","number":317480,"commit":"9a7959558e6deda20f8e917bf41ce30bcc584b64"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/317479","number":317479,"commit":"04bd9d1e71270ea5027a28cc92e14f017c2f7b0a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/315611","number":315611,"commit":"d336259097296a69bb0e639850bd0c1cb7b80105"}],"number":317480}
buildkite-pr-comment-->

nickofthyme committed:
test: update case when `onTitleDuplicate` is missing

nickofthyme committed:
Merge branch 'main' into rm-so-client-lens

nickofthyme committed:
Merge branch 'main' into rm-so-client-lens

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/318499)
* Commit: aa88c0c3f91c75a3d1e98545e3f72e0c7b625421

### Failed CI Steps
* [FTR Configs #6](https://buildkite.com/elastic/kibana-pull-request/builds/318499#0198063e-b571-46f1-8f02-5d8b93b4d038)
* [FTR Configs #6](https://buildkite.com/elastic/kibana-pull-request/builds/318499#01980669-9885-481e-a37c-57593f14e827)
* [FTR Configs #126](https://buildkite.com/elastic/kibana-pull-request/builds/318499#0198063e-b5d2-464f-a120-f1256cf68214)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/318499#0198063e-b5d2-464f-a120-f1256cf68214) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/318499/jobs/0198063e-b5d2-464f-a120-f1256cf68214/artifacts/0198065e-8d49-4b14-b367-796068b3d05d) FTR Configs #<span></span>126 / discover - group 1 Discover CSV Export Generate CSV: archived search "after each" hook for "generates a report with data"
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/318499#0198063e-b571-46f1-8f02-5d8b93b4d038) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/318499/jobs/0198063e-b571-46f1-8f02-5d8b93b4d038/artifacts/01980669-7ec4-455d-812f-7ff9455d9c28) FTR Configs #<span></span>6 / Embeddable Alerts Table Embeddable alerts panel "before all" hook for "should only show alerts from the observability area (o11y+stack) when selecting it"
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/318499#01980669-9885-481e-a37c-57593f14e827) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/318499/jobs/01980669-9885-481e-a37c-57593f14e827/artifacts/01980672-3781-4b47-97e7-c11112252a7c) FTR Configs #<span></span>6 / Embeddable Alerts Table Embeddable alerts panel "before all" hook for "should only show alerts from the observability area (o11y+stack) when selecting it"

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/318499","number":318499,"commit":"aa88c0c3f91c75a3d1e98545e3f72e0c7b625421"}],"number":318499}
buildkite-pr-comment-->

dej611 committed:
Merge branch 'main' into rm-so-client-lens

markov00 commented on src/platform/plugins/shared/visualizations/common/content_management/v1/types.ts:
```diff
@@ -40,6 +34,7 @@ export type VisualizationSavedObjectAttributes = {
   visState?: string;
   uiStateJSON?: string;
   savedSearchRefName?: string;
+  typeName?: string;
```
where this came from?

markov00 commented on src/platform/plugins/shared/visualizations/public/utils/saved_visualize_utils.ts:
```diff
@@ -195,7 +183,10 @@ export async function findListItems(
       if (config) {
         return {
-          ...config.toListItem(savedObject as any),
+          // TODO: understand why this SO can take any shape based on type?
+          // This conflicts with the type of `savedObject` value as `VisualizationSavedObject`.
+          // See test case titled 'uses type-specific toListItem function, if available'
+          ...config.toListItem(savedObject),
```
Do we want to open an issue for that? or is something you want to investigate more in this PR?

markov00 reviewed (commented):
Huge refactoring/cleanup work Nick! Just did a code review but seems pretty clean to me

nickofthyme committed:
chore: update type import

Co-authored-by: Marco Vettorello <vettorello.marco@gmail.com>

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/319485)
* Commit: bd69b25ef4b7f7cfd9f1f9cee188b6baa8821530

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/3a05440803ad90de29e7ebc605068337dfebc071) | [after](https://github.com/elastic/kibana/commit/bd69b25ef4b7f7cfd9f1f9cee188b6baa8821530) | diff |
| --- | --- | --- | --- |
| `lens` | 1422 | 1418 | -4 |
| `visualizations` | 559 | 558 | -1 |
| total |  |  | -5 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/3a05440803ad90de29e7ebc605068337dfebc071) | [after](https://github.com/elastic/kibana/commit/bd69b25ef4b7f7cfd9f1f9cee188b6baa8821530) | diff |
| --- | --- | --- | --- |
| `graph` | 370.6KB | 370.5KB | -53.0B |
| `lens` | 1.5MB | 1.5MB | -1.1KB |
| `visualizations` | 343.5KB | 343.0KB | -569.0B |
| total |  |  | -1.7KB |

#### Public APIs missing exports
> Total count of every type that is part of your API that should be exported but is not. This will cause broken links in the API documentation system. Target amount is 0. `Run node scripts/build_api_docs --plugin [yourplugin] --stats exports` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/3a05440803ad90de29e7ebc605068337dfebc071) | [after](https://github.com/elastic/kibana/commit/bd69b25ef4b7f7cfd9f1f9cee188b6baa8821530) | diff |
| --- | --- | --- | --- |
| `visualizations` | 23 | 22 | -1 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/3a05440803ad90de29e7ebc605068337dfebc071) | [after](https://github.com/elastic/kibana/commit/bd69b25ef4b7f7cfd9f1f9cee188b6baa8821530) | diff |
| --- | --- | --- | --- |
| `lens` | 59.3KB | 59.2KB | -6.0B |
| `visualizations` | 37.6KB | 37.5KB | -84.0B |
| total |  |  | -90.0B |

<details><summary>Unknown metric groups</summary>

#### References to deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/3a05440803ad90de29e7ebc605068337dfebc071) | [after](https://github.com/elastic/kibana/commit/bd69b25ef4b7f7cfd9f1f9cee188b6baa8821530) | diff |
| --- | --- | --- | --- |
| `eventAnnotation` | 6 | 1 | -5 |
| `graph` | 63 | 44 | -19 |
| `lens` | 76 | 5 | -71 |
| `visTypeTimeseries` | 24 | 22 | -2 |
| `visualizations` | 53 | 8 | -45 |
| total |  |  | -142 |


</details>


### History
* :yellow_heart: [Build #318875](https://buildkite.com/elastic/kibana-pull-request/builds/318875) was flaky 13cab6e90c1023accde32e4ecd88f427deb174c6
* :broken_heart: [Build #318499](https://buildkite.com/elastic/kibana-pull-request/builds/318499) failed aa88c0c3f91c75a3d1e98545e3f72e0c7b625421

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/319485","number":319485,"commit":"bd69b25ef4b7f7cfd9f1f9cee188b6baa8821530"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/318875","number":318875,"commit":"13cab6e90c1023accde32e4ecd88f427deb174c6"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/318499","number":318499,"commit":"aa88c0c3f91c75a3d1e98545e3f72e0c7b625421"}],"number":319485}
buildkite-pr-comment-->

markov00 commented on x-pack/platform/plugins/shared/lens/public/app_plugin/save_modal_container.tsx:
```diff
@@ -297,9 +294,8 @@ export const runSaveLensVisualization = async (
-    // this is a lower level call that the Lens attribute service one
-    // @TODO: check if it's worth to replace it witht he attribute service one
-    await checkForDuplicateTitle(
+    // TODO: check if it's worth to replace it with the attribute service one
```
Since we are now using `lensDocumentService.checkForDuplicateTitle` also in the `LensAttributesService` we can remove this comment

nickofthyme replied:
Removed in 1f62a53adeab8b4f9d48c46c2681c43d6f7d0ca4

markov00 reviewed (approved):
LGTM, thanks for this great cleanup!

nickofthyme committed:
chore: remove outdated comment

nickofthyme committed:
Merge branch 'main' into rm-so-client-lens

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/321002)
* Commit: 400d5bc0a9698e9ef0320d4fd4f9f582c39c8b0e

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/f97a952b258cf71c892dd9a72ecb60ac77b49018) | [after](https://github.com/elastic/kibana/commit/400d5bc0a9698e9ef0320d4fd4f9f582c39c8b0e) | diff |
| --- | --- | --- | --- |
| `lens` | 1427 | 1423 | -4 |
| `visualizations` | 547 | 546 | -1 |
| total |  |  | -5 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/f97a952b258cf71c892dd9a72ecb60ac77b49018) | [after](https://github.com/elastic/kibana/commit/400d5bc0a9698e9ef0320d4fd4f9f582c39c8b0e) | diff |
| --- | --- | --- | --- |
| `graph` | 370.9KB | 370.9KB | -53.0B |
| `lens` | 1.5MB | 1.5MB | -1.1KB |
| `visualizations` | 351.3KB | 350.7KB | -569.0B |
| total |  |  | -1.7KB |

#### Public APIs missing exports
> Total count of every type that is part of your API that should be exported but is not. This will cause broken links in the API documentation system. Target amount is 0. `Run node scripts/build_api_docs --plugin [yourplugin] --stats exports` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/f97a952b258cf71c892dd9a72ecb60ac77b49018) | [after](https://github.com/elastic/kibana/commit/400d5bc0a9698e9ef0320d4fd4f9f582c39c8b0e) | diff |
| --- | --- | --- | --- |
| `visualizations` | 23 | 22 | -1 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/f97a952b258cf71c892dd9a72ecb60ac77b49018) | [after](https://github.com/elastic/kibana/commit/400d5bc0a9698e9ef0320d4fd4f9f582c39c8b0e) | diff |
| --- | --- | --- | --- |
| `lens` | 58.9KB | 58.9KB | +12.0B |
| `visualizations` | 37.9KB | 37.8KB | -84.0B |
| total |  |  | -72.0B |

<details><summary>Unknown metric groups</summary>

#### References to deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/f97a952b258cf71c892dd9a72ecb60ac77b49018) | [after](https://github.com/elastic/kibana/commit/400d5bc0a9698e9ef0320d4fd4f9f582c39c8b0e) | diff |
| --- | --- | --- | --- |
| `eventAnnotation` | 6 | 1 | -5 |
| `graph` | 63 | 44 | -19 |
| `lens` | 71 | 4 | -67 |
| `visTypeTimeseries` | 24 | 22 | -2 |
| `visualizations` | 53 | 8 | -45 |
| total |  |  | -138 |


</details>


### History
* :green_heart: [Build #319485](https://buildkite.com/elastic/kibana-pull-request/builds/319485) succeeded bd69b25ef4b7f7cfd9f1f9cee188b6baa8821530
* :yellow_heart: [Build #318875](https://buildkite.com/elastic/kibana-pull-request/builds/318875) was flaky 13cab6e90c1023accde32e4ecd88f427deb174c6
* :broken_heart: [Build #318499](https://buildkite.com/elastic/kibana-pull-request/builds/318499) failed aa88c0c3f91c75a3d1e98545e3f72e0c7b625421

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/321002","number":321002,"commit":"400d5bc0a9698e9ef0320d4fd4f9f582c39c8b0e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/319485","number":319485,"commit":"bd69b25ef4b7f7cfd9f1f9cee188b6baa8821530"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/318875","number":318875,"commit":"13cab6e90c1023accde32e4ecd88f427deb174c6"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/318499","number":318499,"commit":"aa88c0c3f91c75a3d1e98545e3f72e0c7b625421"}],"number":321002}
buildkite-pr-comment-->
